pkg/yamlprocessor/yamlprocessor.go

```diff
@@ -104,5 +104,11 @@ func (p *Processor) ReplaceString(path, value string) error {
 }
 
 func (p *Processor) Bytes() []byte {
-	return []byte(p.file.String())
+	str := p.file.String()
+
+	if len(str) > 0 && str[len(str)-1] != '\n' {
```

Comment:
Could you add a reminder here (TODO or NOTE) to remove this once github.com/goccy/go-yaml/pull/329 is merged and released 😄

Reply:
updated https://github.com/pipe-cd/pipecd/pull/4033/commits/fabf67ee34c041c22e72f36a660f2dcbc1d90931

Reply:
@khanhtc1202 I updated go-yaml and removed TODO.